✨ Add MachinePool support for FailureDomains (#3157)
kubernetes-sigs/cluster-api: docs/proposals/20190919-machinepool-api.md

  - **ClusterName [required]**
    - Type: `string`
    - Description: Name of the Cluster this machine pool belongs to.
  - **FailureDomains [optional]**
    - Type: `[]string`

>     - Description: FailureDomainSpec is the Schema for Cluster API failure domains.
>       It allows controllers to understand how many failure domains a cluster can optionally span across.

Reviewer: ```suggestion
    - Description: FailureDomains is the list of failure domains this MachinePool should be attached to.
```

>   - **FailureDomains [optional]**
>     - Type: `FailureDomains`
>     - Description: FailureDomainSpec is the Schema for Cluster API failure domains.
>       It allows controllers to understand how many failure domains a cluster can optionally span across.

Reviewer: ```suggestion
  - **FailureDomains [optional]**
    - Type: `FailureDomains`
    - Description: FailureDomains are the failure domains the MachinePool instances should be be created in.
```
